hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java

```diff
@@ -293,7 +293,19 @@ public void updateBackupInfo(BackupInfo info) throws IOException {
     }
     try (Table table = connection.getTable(tableName)) {
       Put put = createPutForBackupInfo(info);
-      table.put(put);
+      try {
+        table.put(put);
+      } catch (Exception e) {
+        // If the BackupInfo update can't be processed, then we should fall back to
+        // the previous BackupInfo, but also update it to reflect the failure.
+        LOG.error("Failed to update BackupInfo for {}. Marking as failed", info.getBackupId(), e);
+        BackupInfo legacyInfo = readBackupInfo(info.getBackupId());
+        if (legacyInfo != null) {
+          legacyInfo.setFailedMsg("Failed to update BackupInfo. Error: " + e.getMessage());
+          table.put(createPutForBackupInfo(legacyInfo));
```

Comment:
Why write back the same info that was just read? Or does the new `put` include new information in addition to the content of `legacyInfo`? Will this be confusing to an operator, to see a melange of both?

Reply:
It might be — I think the implication of this codepath is that we've lost the new update, so we need to rely on our legacy update (plus the failedMsg)

Reply:
I see that you're calling set**Failed**Msg, which i guess implies setting a failed status? I missed that detail on my first review.

What are the odds that a second Put call will succeed where the first failed? If we expect that a second attempt at the Put might succeed, we could instead wrap sending the original Put in a retry loop... though I suppose the client itself will make some retry attempts.

Do we have no other means of marking the backup as having failed? Because the result of a failure here is dataloss over subsequent backups, perhaps we need a more robust system for identifying failures, to be cautious at several logical checkpoints. For example, maybe we can detect the absence of some value to recongize a failure? @DieterDP-ng, @hgromer : I'd love to hear your thoughts here as well.

Anyway, this thread of thought is expanding beyond the scope of the ticket you've filed. If you want to just push the truncation change, or the truncation change plus a retry loop on the original Put, we could leave the rest for a larger issue.

Reply:
That's all correct. FWIW, we do have other means for identifying backup failures — without a success marker file I believe this backup could not be used to restore a table. The problem is that we have multiple sources of truth for whether a backup has succeeded — the marker file and entries in the system table. Missing updates in the system table cause failures that are only really salvageable by purging the table and/or taking a new full backup

Reply:
I'm happy to just remove this error handling too though, I was on the fence about it's addition and did, when digging into this, wish that we had tried harder to ensure missed BackupInfo updates were documented for the system's future reference

Reply:
I don't think adding this logic here will have any noticable effect. Since an Exception is thrown in this method, calling code will most likely end up in `TableBackupClient#failBackup`, which will overwrite the failure message (with the same message: the one from the Exception).

So in the interest of keeping things simple, I'm pro reverting this specific change. The limiting of the error message length should already ensure a proper error gets stored.

I wouldn't oppose some kind of retry mechanism for failed PUTs, but I'd expect that connection issues are automatically retried when executing PUTs, and a content-based PUT failure will still fail when retried.